docs/spec/broker.md

```diff
@@ -44,6 +44,14 @@ The attributes filter specifying a list of key-value pairs MUST be supported by
 Trigger. Events that pass the attributes filter MUST include context or
 extension attributes that match all key-value pairs exactly.
 
+### Delivery Spec
+
+Both BrokerSpec and TriggerSpec have a `Delivery` field of type [`duck.DeliverySpec`](./spec.md#eventingduckv1deliveryspec).
+This field, among the other features, allows the user to define the dead letter sink and retries.
+The `BrokerSpec.Delivery` field is global across all the Triggers registered with that particular
+Broker, while the `TriggerSpec.Delivery`, if configured, fully overrides `BrokerSpec.Delivery` for
+that particular Trigger.
+
```

Comment:
I just want to understand how this is supposed to work with the wording about defaulting fields. It's more of an implementation question but has implications on when the defaulting happens and if it even should happen. The PR description seems to say that we default the triggerspec.delivery field, so if that's how we're going to implement it then it matters when the defaulting applies. If however you implement it by not actually updating the triggerspec.delivery but look at the brokerspec.delivery if triggerspec.delivery is missing then the behaviour is different.

In particular, it would be good to clarify that, for example, what happens in the following scenarios:
1. Create Broker, no delivery
2. Create Trigger, no delivery
3. Update Broker, configure delivery, what happens here?

1. Create Broker, with delivery
2. Create Trigger, no delivery -> defaulting happens
3. Update Broker, delete/update broker delivery -> what happens to trigger above
4. Create Trigger2, no delivery, will it match what Trigger from 2 has?

Just want to understand that behaviour little better about what "defaulting" really means and what the expected behaviour is.
As I said, depending on how this is implemented and if we actually modify the triggerspec.delivery vs. treating missing triggerspec.delivery as 'use brokerspec.delivery' has different semantics potentially and certainly makes the implementation more tricky depending on how it's implemented.

Reply:
From the PR description, this is what I'm trying to understand with the description on the spec.
```
Now Broker.Delivery defaults the Trigger.Delivery field. Note: this is a breaking change for mtbroker users that are using the Broker.Delivery field, because this field, even if it remains the same, will change meaning for them (in a followup I'll perform the actual change)
```

Reply:
Defaulting basically means that the trigger delivery spec, if any, wins on the broker delivery spec, exactly as it works today with subscription and channel delivery fields. So the result of the above cases:

2. defaults to Broker.Delivery -> no delivery
3. defaults to Broker.Delivery -> delivery from Broker.Delivery

5. defaults to Broker.Delivery -> delivery from Broker.Delivery
6. defaults to Broker.Delivery -> delivery from Broker.Delivery (no delivery if Broker.Delivery was removed)
7. defaults to Broker.Delivery -> delivery from Broker.Delivery (no delivery if Broker.Delivery was removed)

Another case:

8. Create Trigger3, with delivery -> delivery from Trigger3.Delivery

Reply:
I can provide the implementation for mtbroker to show how that works, but I don't see any particular implementation issues with it

Reply:
Right, so I think it was (as I tried to say) how it's worded about "defaulting" since typically defaulting means filling in the fields and if that's how you were going to implement it, it could get confusing. If you remove the word defaulting I think I'd be much happier. What I tried to say :) was that spec seems ok, but your PR description confused me wrt. using words defaulting.
If you say:
If triggerspec.delivery is specified it is used. If triggerspec.delivery does not exist brokerspec.delivery is used. Nothing about defaulting :)

Reply:
How about now? I removed the default word everywhere and added some wording to the spec